src/App.java

```diff
@@ -1,5 +1,595 @@
+                Pattern recordPattern = Pattern.compile(
+                    "\\{\\s*\"title\"\\s*:\\s*\"([^\"]+)\"\\s*," +
+                    "\\s*\"checkOutDate\"\\s*:\\s*\"(\\d{4}-\\d{2}-\\d{2})\"\\s*," +
+                    "\\s*\"dueDate\"\\s*:\\s*\"(\\d{4}-\\d{2}-\\d{2})\"\\s*," +
+                    "\\s*\"returnedDate\"\\s*:\\s*(?:\"(\\d{4}-\\d{2}-\\d{2})\"|(null))" +
+                    "\\s*\\}",
+                    Pattern.DOTALL
+                );
+
+                // The matcher will look for the pattern in the string and return the first match it finds.
+                Matcher recM = recordPattern.matcher(booksBlock);
+                while (recM.find()) {
+                    String title = recM.group(1); // Book title
+                    LocalDate checkout = LocalDate.parse(recM.group(2)); // checkout date
+                    LocalDate due = LocalDate.parse(recM.group(3)); // due date
+                    String retString = recM.group(4); // returned date
+                
+                    LocalDate returned;
+                    if (retString != null) {
+                        returned = LocalDate.parse(retString); // convert from string to LocalDate
+                    } else {
+                        returned = null;
+                    }
+                
+                    Book book = library.findBookByTitle(title);
+                    if (book != null) {
+                        book.setCheckedOut(returned == null);
+                        book.setStudentId(studentId);
+                        student.addLoanRecord(book, checkout, due, returned);
+                    }
+                }
+            }
+        } catch (FileNotFoundException e) {
+            printer.output("Error: File not found.");
+            e.printStackTrace();
+        }
+    
+        return library;
+    }    
+
+    /**
+     * Displays the title of the program in a cool way. I separated this out so I can reuse the menu without displaying the title in the future.
+     */
+    public void displayTitle() { 
+        printer.output("\n" +
+                        "████████╗██╗░░██╗███████╗  ██╗░░░░░██╗██████╗░██████╗░░█████╗░██████╗░██╗░░░██╗\n"+
+                        "╚══██╔══╝██║░░██║██╔════╝  ██║░░░░░██║██╔══██╗██╔══██╗██╔══██╗██╔══██╗╚██╗░██╔╝\n"+
+                        "░░░██║░░░███████║█████╗░░  ██║░░░░░██║██████╦╝██████╔╝███████║██████╔╝░╚████╔╝░\n"+
+                        "░░░██║░░░██╔══██║██╔══╝░░  ██║░░░░░██║██╔══██╗██╔══██╗██╔══██║██╔══██╗░░╚██╔╝░░\n"+
+                        "░░░██║░░░██║░░██║███████╗  ███████╗██║██████╦╝██║░░██║██║░░██║██║░░██║░░░██║░░░\n"+
+                        "░░░╚═╝░░░╚═╝░░╚═╝╚══════╝  ╚══════╝╚═╝╚═════╝░╚═╝░░╚═╝╚═╝░░╚═╝╚═╝░░╚═╝░░░╚═╝░░░");
+    }
+
+    /**
+     * Displays the main menu.
+     * Most of its functionality is handling user input and calling the appropriate methods in the Library class.
+     */
+    public void displayMenu() {

```

Comment:
this is a huge method! There's no hard and fast rule about how long a method can be before you try to break into smaller methods, but this is well past whatever the suggested length is. You should make each of the subcases its own submethod.